gopages/internal/generate/generate.go

```diff
@@ -158,7 +158,7 @@ var crawlerStaticPipe = pipe.New(pipe.Options{}).
 	}).
 	Append(func(args docsArgs) (docsArgs, error) {
 		// Generate main index to redirect to actual content page. Important to separate from 'lib' top-level dir.
-		return args, util.WriteFile(args.FS, filepath.Join(args.OutputPath, "index.html"), []byte(redirect("pkg/"+args.ModulePackage)), scrapeFilePermission)
+		return args, util.WriteFile(args.FS, filepath.Join(args.OutputPath, "index.html"), []byte(redirect("pkg/"+args.ModulePackage+"/")), scrapeFilePermission)
```

Comment:
This can cause extra redirects (`/foo/` -> `/foo`) on some HTTP servers. Is this change required to fix your code's doc links? If so, do you have an example scenario to help reproduce it?

Reply:
All package & subpackage links are relative links.

When I browse to root `/<base>/`, I get redirected to `/<base>/<mymodule>`. From that page, when I click on a package I'm sent to `/<base>/<mypackage>` instead of `/<base>/<mymodule>/<mypackage>`.
Forcing a trailing slash fixes that issue.

If a server redirects from `/foo/` to `/foo` then it would always render pages with broken links.

I totally agree it would be much better if the server I'm deploying too was configured to perform such redirection, unfortunately I'm not managing this server and cannot change its config. I can revert this change.

Another approach would be to change all the package links to make them absolute instead of relative.

Reply:
Thinking about the extra-redirects removing the trailing slash. The servers doing that are likely to serve pages with broken links anyway. And all links to subpackages have a trailing slash already. So doing sunch kind of redirect in the root index.html would be more consistent with how package  links are rendered evrywhere else

Reply:
Hmm I'm trying to understand your last comment, though I'm not 100% sure what you mean. Do you have a docs site up that triggers this issue I could take a poke at? Or maybe a minimal example code with which gopages generates the issue?
If I'm understanding correctly, sounds like the package index URLs are badly formed – just not sure I understand how it would happen.

I'd be happy to explore this further, too. Maybe open a new issue to discuss?